File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> result;
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<int> current;
        backtrack(nums, target, current, 0);
        return result;
    }
private:
    void backtrack(vector<int>& nums, int target, vector<int>& current, int i){
        if (target == 0){
            result.push_back(current);
            return;
        }
        if (target < 0 || i >= nums.size()){
            return;
        }
        current.push_back(nums[i]);
        backtrack(nums, target - nums[i], current, i);
        current.pop_back();
        backtrack(nums, target, current, i + 1);

    }
};

File: Data Structures & Algorithms/concatenation-of-array/submission-0.py
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        lans=[]
        for i in range(len(nums)*2):
            if i >= len(nums):
                lans.append(nums[i-len(nums)])
            else:
                lans.append(nums[i])

        return lans

            
